src/factors/Factors.jl

```diff
@@ -33,6 +33,105 @@ function Factor(val; kwargs...)::Param
     return Param((; val=val, nt...))
 end
 
+struct CategoricalDistribution{T}
+    categories::Vector{T}
+    distribution::Categorical
+end
+
+"""
+    CategoricalDistribution(categories::Vector{T}, weights::Vector{Float64})::CategoricalDistribution{T} where {T}
+    CategoricalDistribution(categories::Vector{T})::CategoricalDistribution{T} where {T}
+    CategoricalDistribution(categories...)::CategoricalDistribution
+
+Construct a categorical variable. Default to a uniform cateforical variable if the 
+probability weightings are not provided.
+"""
+function CategoricalDistribution(
+    categories::Vector{T}, weights::Vector{Float64}
+)::CategoricalDistribution{T} where {T}
+    if length(unique(categories)) != length(categories)
+        throw(ArgumentError("Categories in a categorical variable must be unique."))
+    end
+    if length(categories) != length(weights)
+        msg = "Length of categories and weightings do not match."
+        msg *= " Got $(length(categories)) and $(length(weights))."
+        throw(ArgumentError(msg))
+    end
+    if !(sum(weights) ≈ 1) && all(weights .>= 0.0)
+        throw(ArgumentError("Weights must sum to one."))
+    end
+    return CategoricalDistribution(
+        categories,
+        Categorical(weights))
```

Comment:
Minor nitpick but I prefer spacing for readability

```suggestion
    if length(unique(categories)) != length(categories)
        throw(ArgumentError("Categories in a categorical variable must be unique."))
    end

    if length(categories) != length(weights)
        msg = "Length of categories and weightings do not match."
        msg *= " Got $(length(categories)) and $(length(weights))."
        throw(ArgumentError(msg))
    end

    if !(sum(weights) ≈ 1) && all(weights .>= 0.0)
        throw(ArgumentError("Weights must sum to one."))
    end

    return CategoricalDistribution(
        categories,
        Categorical(weights)
    )
```